packages/auth-construct/API.md

```diff
@@ -47,7 +49,7 @@ export type AuthProps = {
         externalProviders?: ExternalProviderOptions;
     };
     senders?: {
-        email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'>;
+        email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | IFunction | AmplifyFunction;

```

Comment:
`AmplifyFunction` should be an input to `defineAuth` not to the construct.

There are couple of examples is `backend-data` that may be looked at.
For example, take a look how these two are handled 
1. https://github.com/aws-amplify/amplify-backend/blob/6e4a62f1d0fe98e4a96cdc44c6d6b40cd9269397/packages/backend-data/src/types.ts#L41
2. https://github.com/aws-amplify/amplify-backend/blob/6e4a62f1d0fe98e4a96cdc44c6d6b40cd9269397/packages/backend-data/src/types.ts#L141


It should be that

```
const someFunction = defineFunction();
const auth = defineAuth({ senders: { email: someFunction });
```


under the hood. the auth construct factory should resolve the reference to IFunction (see how data does this under the hood) and pass it to auth construct as `IFunction`.